### PR TITLE
Use function to open a sidebar file

### PR DESCRIPTION
**Pull request by skjnldsv**
Capture d’écran_2019-10-30_12-00-35

### CODE REVIEW

**Comment by danxuliu** [comment]
>  But I think it's because we indeed remove the app-sidebar now, so it's ok?

Yes, it is fine; if you see `Trying again` there is no problem ;-)

_But there is a failure in the tests!_ Yes, but the failure is in a different scenario than the one that you copied :-P That error is unrelated to this pull request, is a flaky test that I have seen from time to time. I will fix that one and some other flaky tests in a different pull request ;-)

File: apps/files/js/fileactions.js [hunk 1]
```diff
@@ -708,7 +708,7 @@
 					// close sidebar on delete
 					const path = context.dir + '/' + fileName
 					if (OCA.Files.Sidebar && OCA.Files.Sidebar.file === path) {
-						OCA.Files.Sidebar.file = undefined
```
Comment (ChristophWurst) [inline apps/files/js/fileactions.js]: this

File: apps/files/js/filelist.js [hunk 2]
```diff
@@ -630,7 +630,7 @@
 			}
 
 			if (!fileName) {
-				OCA.Files.Sidebar.file = null
```
Comment (ChristophWurst) [inline apps/files/js/filelist.js]: this

File: apps/files/src/views/Sidebar.vue [hunk 3]
```diff
@@ -261,7 +261,7 @@ export default {
 		},
 		onClose() {
 			this.resetData()
-			OCA.Files.Sidebar.file = ''
```
Comment (ChristophWurst) [inline apps/files/src/views/Sidebar.vue]: and this shows why the `close` method is more appropriate here :D

**Review by ChristophWurst — APPROVED** [review]

**Comment by rullzer** [comment]
conflicts!

**Comment by skjnldsv** [comment]
@rullzer I need reviews :p

**Review by juliusknorr — APPROVED** [review]
Looks good to me now :+1:

**Pull request merged**